app/src/main/java/Models/DBManager.java

```diff
@@ -95,4 +101,55 @@ public boolean checkForUser(User u)
         return valid;
 
     }
+
+    //Product
+    public void addProduct(String name, double price, String description )
+    {
+
+        SQLiteDatabase db = this.getWritableDatabase();
+        ContentValues vals = new ContentValues();
+        vals.put(Product.PRODUCT_COL_NAME,name);
+        vals.put(Product.PRODUCT_COL_PRICE,price);
+        vals.put(Product.PRODUCT_COL_DESCRIPTION,description);
+        db.insert(Product.PRODUCT_TAB_NAME,null, vals);
+        db.close();
+    }
+
+    public ArrayList<Product> getProductList()
+    {
+        ArrayList<Product> list = new ArrayList<Product>();
+        SQLiteDatabase db = this.getWritableDatabase();
+
+        String GET_LIST = "SELECT * FROM " + Product.PRODUCT_TAB_NAME+";";
+        Cursor c = db.rawQuery(GET_LIST,null);
+
+        if (c.moveToFirst())
+        {
+            do {
+                Product prod = new Product();
+                prod.setId(c.getInt(c.getColumnIndex(Product.PRODUCT_COL_ID)));
+                prod.setName(c.getString(c.getColumnIndex(Product.PRODUCT_COL_NAME)));
+                prod.setPrice(c.getDouble(c.getColumnIndex(Product.PRODUCT_COL_PRICE)));
+                prod.setDescription(c.getString(c.getColumnIndex(Product.PRODUCT_COL_DESCRIPTION)));
+                list.add(prod);
+            } while (c.moveToNext());
+        }
+        return list;
+    }
+
+    public boolean productExists(Product p)
+    {
+        ArrayList<Product> list = getProductList();
+        boolean exists = false;
+        for (int i = 0; i < list.size(); i++)
+        {
+            if (p.getName().equals(list.get(i).getName()))
+            {
+                exists = true;
```

Comment:
would remove the variable exists, make it `return true;` and `return false;` after the loop instead of `return exists;`